### ISSUE TITLE
Require the developer to activate textScaleFactor 

### ISSUE THREAD

**Issue by jhsware** [flutter/flutter#41100]
On iOS, the automatic enforcement of system wide textScaleFactor leads to unexpected and potentially breaking failure of an app layout.

It would be better if textScaleFactor was explicitly activated by the developer, who then has a chance of investigating layout side-effects.

Alternatively it should be added to a pre-launch testing check-list so developers have a chance to react to this behaviour before it impacts the users.

**Comment by HansMuller** [comment]
Are there are specific components or layouts that are failing when textScaleFactor gets large?

**Comment by jhsware** [comment]
It is the developer generated layout. For responsive designs you obviously need to consider how text behaves when the user changes it's size. Also, some designs just aren't responsive because you have maximised the use of available space.

Just released an app and started getting hundreds of support tickets with images like this:

![image](https://user-images.githubusercontent.com/3896884/65491974-63b61b00-deb0-11e9-8485-01b7f9805a1c.png)

What happened is that the users have textScaleFactor way upp because of poor sight. This would have been fine if we had known about this feature. But because it is on by default and not widely discussed, it comes as an unpleasant surprise.

I believe you will see the same effect for both Cupertino and Material design apps. It is more a "Media query" issue.

**Comment by athornz** [comment]
This definitely shouldn't be an opt-in feature - text scaling should be on by default so users with limited sight can still use apps, even if they are in a slightly broken state.

Android apps have had text scaling from day 1, it's just one of those things you need to test.

**Comment by jhsware** [comment]
The massive amounts of feedback I have received from users with limited sight indicates that they don't agree with you. They would rather have an app in working state than the ability to force-zoom the text. 

The feedback I got from these users is that the app is broken and once the font size was reverted to the correct (smaller) size they where happy. In their opinion, slightly broken = useless state.

So I would say that real world user feedback supports making this opt-in.

That said I laud the principal and the alternative is to update the documentation to focus on building responsive designs.

**Comment by csells** [comment]
@jhsware do you have an idea of how we could make this automatic behavior more explicit for Flutter developers such that they remember to include this in their testing?

**Comment by jhsware** [comment]
@csells Perhaps you could add a section to `pubspec.yaml` something like:

```yaml
# The following section is specific to Flutter.
flutter:
  uses-material-design: true

  # Text scaling is controlled by the host OS. This is an accessability feature
  # and enabled by default. For more info and how to disable, see:
  # https://flutter.dev/text-scale-factor
  disable-text-scale-factor: false 

  # For details regarding adding assets from package dependencies, see
  # https://flutter.dev/assets-and-images/#from-packages
  assets:
    - assets/images/
```

As a new user you check this part out and then you can learn more about this feature by visiting the URL. On the landing page you can note that the disable-text-scale-factor toggle doesn't have any effect but that there are work arounds. There you can also discuss testing.

This way you don't need any code changes but it is a lot harder to miss.

**Comment by goderbauer** [comment]
Allowing developers to disable text scaling like that would be against accessibility guidelines.
